src/expr.rs

```diff
@@ -245,8 +245,8 @@ fn format_expr(expr: &ast::Expr,
         ast::ExprKind::InlineAsm(..) => {
             wrap_str(context.snippet(expr.span), context.config.max_width, shape)
         }
-        // FIXME(#1537)
-        ast::ExprKind::Catch(..) => unimplemented!(),
+        // 6 = `catch `
+        ast::ExprKind::Catch(ref block) => block.rewrite(&context, try_opt!(shape.sub_width(6))),
```

Comment:
Yeah, that is correct (and unfortunate). I'm not sure how to best address it, to be honest. Perhaps pull the code out of the Rewrite impl and into a free function with an extra flag?